text/0000-export-to-oci.md

```diff
@@ -0,0 +1,715 @@
+
+  Some thoughts about this approaches
+
+    - **Process Management:** Platforms must now manage a parallel process (registry in the daemon). This would entail ensuring that the registry is started and cleaned up appropriately.
+    - **Networking:** There are additional network complications in order to route images to the ephemeral registry. For example, [network drivers](https://docs.docker.com/network/#network-drivers), [proxy](https://docs.docker.com/desktop/networking/#httphttps-proxy-support) and [DNS configuration](https://docs.docker.com/config/containers/container-networking/#dns-services), [host name resolution](https://docs.docker.com/desktop/networking/#i-want-to-connect-from-a-container-to-a-service-on-the-host), and [TLS certificates](https://betterprogramming.pub/deploy-a-docker-registry-using-tls-and-htpasswd-56dd57a1215a) to name a few.
+
+- Why is this proposal the best? [OCI Image Layout](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) format is a standard from which other tools can create a [OCI Runtime Specification bundle](https://github.com/opencontainers/runtime-spec/blob/v1.0.0/bundle.md) exporting to this format enables Platforms to implement any feature in the top of this format, for example, exporting to [containerd](https://containerd.io) has been [requested](https://github.com/buildpacks/lifecycle/issues/829) by the community and it could be implemented if we can get the application image exported in [OCI Image Layout](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) format.
+- What is the impact of not doing this? Probably will never remove the Daemon support in the Lifecycle
+
+# Prior Art
+[prior-art]: #prior-art
+
+- Discussion around removing the Daemon support [RFC](https://github.com/buildpacks/rfcs/blob/jjbustamante/feature/deprecate-daemon/text/0000-deprecate-daemon.md)
+
+# Unresolved Questions
+[unresolved-questions]: #unresolved-questions
+
+- What parts of the design do you expect to be resolved before this gets merged?
+  - Tools like [umoci](https://umo.ci/) used to create a runtime bundle from an image in OCI layout format, requires the [annotation](https://github.com/opencontainers/image-spec/blob/main/annotations.md#pre-defined-annotation-keys) `org.opencontainers.image.ref.name` to be present. Also tools like [skopeo](https://github.com/containers/skopeo) when an image is `copy` in oci format the annotation is included.
+  We are not adding the annotation as part of the Buildpacks Specification, but in this case this could make our output incompatible with some other tooling.
+  - Depending on the previous question, the rules for mapping an image reference into a path could change, also validations that the Lifecycle could do to verify the `org.opencontainers.image.ref.name` against the value provided in the image reference.
+  - Exporting to a tarball can be handle on this RFC or a new one must be created?
```

Comment:
I don't think it needs to be tied to this. I think if we decide to write a tar there could be a pack specific RFC  (if even that is needed - is a PR enough for an additive feature?)

Reply:
Well my concern was because of this: 

If we want the image to be exported as a tarball, then the `Image.SaveFile()` must be called, so it means we need to expose in the `exporter` something to do that. I added the flag `-layout` to tell the export the image must be export to OCI layout format, so my concern was, if we want that for a tarball, then something else needs to be expose. That's why I was thinking on a RFC

Reply:
Maybe we could derive the format from the target path (e.g., if it has .tar or .tgz or whatever) or from some prefix (such as `oci-archive:`) but I agree that could be done separately from this RFC.